### ISSUE TITLE
Entering non-breaking space followed by some punctuation marks crashes the document when interface is in French

### ISSUE THREAD

**Issue by Silejonu** [flathub/org.onlyoffice.desktopeditors#65]
ONLYOFFICE version 7.0.1.37 ([bug present on the AppImage version as well](https://github.com/ONLYOFFICE/appimage-desktopeditors/issues/37))

### Steps to reproduce to encounter the bug:
1. Set interface to French
2. Open a new document
3. Type a non-breaking space (` `)
4. Type a punctuation mark among `?`, `!`, `:` and `;`
5. The document crashes

### Additional context:
In French, a non-breaking space must be place before `?`, `!`, `:` and `;`.

When typing a non-breaking space, ONLYOFFICE actually prints a regular space followed by a non-breaking space.

When typing a punctuation mark that needs to be preceded by a non-breaking space, ONLYOFFICE replaces the space before the character by a non-breaking one. When it is preceded by a space + a non-breaking space, it crashes the document. If one types a non-breaking space, deletes the preceding regular space, then proceeds to type one of the problematic characters, the document will **not** crash.

**Comment by ShockwaveNN** [comment]
Hi

> Type a non-breaking space ( )

Is copying this symbol is enough to reproduce your issue, because seems for it's not crashed

Also could you check deb version (which already v7.1.0), maybe we already fixed that issue?

**Comment by Silejonu** [comment]
> Hi
> 
> > Type a non-breaking space ( )
> 
> Is copying this symbol is enough to reproduce your issue, because seems for it's not crashed

As I wrote, if you just remove the regular space that's being typed just before the non-breaking space, it doesn't crash. This is the same thing when copy/pasting the non-breaking space. The issue seems to be about a regular space being added before the non-breaking space when typing a non-breaking space. I hope this makes sense. ^^

> Also could you check deb version (which already v7.1.0), maybe we already fixed that issue?

I checked with the RPM version, and this issue seems to not be present.
